srml/balances/src/lib.rs

```diff
@@ -851,7 +854,12 @@ where
 		}
 	}
 
-	fn transfer(transactor: &T::AccountId, dest: &T::AccountId, value: Self::Balance) -> Result {
+	fn transfer(
+		transactor: &T::AccountId,
+		dest: &T::AccountId,
+		value: Self::Balance,
+		existence_requirement: ExistenceRequirement,
```

Comment:
Not a dev so take this with a grain of salt. The only logic is based around the `KeepAlive` variant, so I wonder if you can make this an optional parameter and then do `if let Some(existence_requirement)`. It would remove the opinion of choosing a default in other places in the codebase.

Reply:
Yeah, that's a fair point. Perhaps we could use `AllowDeath` as the default though?

Reply:
This would make it also compatible with all the old transactions I believe since `encode(None) == []`, hence the new format can be decoded from the old bytes.

Reply:
None is does not encode to an empty array.

Reply:
right it is `0x00`. nevermind this then :D

Reply:
@jacogr How does polkadot.js handles missing trailing parameter on a dispatchable call? Does it throw or adds a default value? For later case, make `existence_requirement: Option< ExistenceRequirement>` will be a tolerable change.

Reply:
It expects all params, so all calls will fail when this is introduced. Even if did unexpected things and added a default value, that would not make sense unless it is an `Option` - in this case it would make sense "not to break". (Not the case, only length check atm, so all expected)

(There are no completely empty types at all, `Option` also needs a single byte to encode none)